test/test_boundary_conditions_integration.jl

```diff
@@ -160,6 +164,67 @@ function test_flat_extrapolation_open_boundary_conditions(arch, FT)
     end
 end
 
+wall_normal_boundary_condition(::Val{1}, obc) = (; u = FieldBoundaryConditions(east = obc, west = obc))
+wall_normal_boundary_condition(::Val{2}, obc) = (; v = FieldBoundaryConditions(south = obc, north = obc))
+wall_normal_boundary_condition(::Val{3}, obc) = (; w = FieldBoundaryConditions(bottom = obc, top = obc))
+
+normal_velocity(::Val{1}, model) = model.velocities.u
+normal_velocity(::Val{2}, model) = model.velocities.v
+normal_velocity(::Val{3}, model) = model.velocities.w
+
+velocity_forcing(::Val{1}, forcing) = (; u = forcing)
+velocity_forcing(::Val{2}, forcing) = (; v = forcing)
+velocity_forcing(::Val{3}, forcing) = (; w = forcing)
+
+function test_pertubation_advection_open_boundary_conditions(arch, FT)
+    for orientation in 1:3
+        topology = tuple(map(n -> ifelse(n == orientation, Bounded, Flat), 1:3)...)
+
+        grid = RectilinearGrid(arch, FT; topology, size = (4, ), x = (0, 4), y = (0, 4), z = (0, 4), halo = (1, ))
+
+        obc = PerturbationAdvectionOpenBoundaryCondition(1, inflow_timescale = 0.1)
+
+        boundary_conditions = wall_normal_boundary_condition(Val(orientation), obc)
+
+        model = NonhydrostaticModel(; grid, boundary_conditions, timestepper = :QuasiAdamsBashforth2)
+
+        u = normal_velocity(Val(orientation), model)
+        
+        CUDA.@allowscalar u.data .= 1
+
+        time_step!(model, 1)
+
+        @test all(u .== 1)
+
+        end_index = tuple(map(n -> ifelse(n == orientation, 5, 1), 1:3)...)
+        
+        CUDA.@allowscalar u[end_index...] = 2
+
+        time_step!(model, 1)
+        
+        CUDA.@allowscalar @test all(interior(u, end_index...) .== 1.5)
```

Comment:
I think it's great that we're testing values, and not just that things compile. However, I don't fully understand why these values are the appropriate ones (mostly referring to 1.5). Could you please add a comment here to explain the reasoning?